File: WordPig/app/src/main/java/com/pigdogbay/wordpig/model/Screen.java
package com.pigdogbay.wordpig.model;

import com.pigdogbay.library.utils.ObservableProperty;

/**
 * Created by Mark on 07/04/2015.
 */
public class Screen
{
    public enum ScreenState
    {
        Home,
        Game,
        GameOver
    }

    public ObservableProperty<ScreenState> screenStateObserver;

    public Screen()
    {
        screenStateObserver = new ObservableProperty<ScreenState>(ScreenState.Home);
    }

}

File: WordPig/app/src/main/java/com/pigdogbay/wordpig/model/Screen.kt
package com.pigdogbay.wordpig.model

import com.pigdogbay.lib.utils.ObservableProperty

/**
 * Created by Mark on 07/04/2015.
 */
class Screen {
    enum class ScreenState {
        Home,
        Game,
        GameOver
    }

    val screenStateObserver = ObservableProperty(ScreenState.Home)
}
